pkg/controller/datadogagent/utils.go

```diff
@@ -528,30 +544,22 @@ func getEnvVarsForAgent(dda *datadoghqv1alpha1.DatadogAgent) ([]corev1.EnvVar, e
 				ValueFrom: getClusterAgentAuthToken(dda),
 			},
 		}
-		if *spec.ClusterAgent.Config.ClusterChecksRunnerEnabled && spec.ClusterChecksRunner == nil {
-			clusterEnv = append(clusterEnv, corev1.EnvVar{
-				Name:  datadoghqv1alpha1.DDExtraConfigProviders,
-				Value: datadoghqv1alpha1.ClusterChecksConfigProvider,
-			})
+		if *spec.ClusterAgent.Config.ClusterChecksEnabled {
+			if spec.ClusterChecksRunner == nil {
```

Comment:
Why do we have a distinction on provider when no CLC?

Reply:
when clc runners disabled the node agent needs to be configured to pull cluster checks AND endpoint checks, when the clc runners enabled they take care of running cluster checks and the node agents must only endpoint checks